airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -159,12 +159,84 @@ def get_start_timestamp(self, stream_state) -> int:
 
         return start_point
 
+class IncrementalStripeStreamWithUpdates(IncrementalStripeStream):
+    """
+        This is a base class for incremental streams that support updates using the events API
+        In first sync it will not get any updates as the records are already updated
+        After first sync it will get all updates starting from date of the last sync
+    """
+    event_types = None
+    update_field = "event_created"
+
+    def read_records(self,stream_slice, stream_state, **kwargs) -> Iterable[Mapping[str, Any]]:
+        # If this is not the first sync also get the updates
+        if stream_state.get(self.cursor_field) is not None: 
```

Comment:
`if stream_state.get(self.cursor_field) is not None:` This might be one buggy scenario. Say first sync saved state tille some time T and before finished something crashed. We might assume, next task should read from event stream.
In blendo, not sure how it was handled, probably for first sync only after fetching all data, state was being saved.

I think there is no easy way to solve this and we can take this as separate backlog task. In rudder-sources we will need to have code for these specific stripe resources if first sync fails, clean up WH table data and reset the state.

Reply:
To get this more clear on how this works using subscriptions as example
1. In first sync we get the subscriptions, ignoring the events
2. In all other syncs we first get all the new subscriptions using the original state, and then we get the updated subscriptions using the events, and we keep only the events that are not part of the new subscriptions we just fetch because are already updated.

Does this make sense?

Reply:
@ssbeefeater Yes. I might be  understanding something wrong. I was thinking of this scenario and how are we checking its a first sync for subscription.

1. First sync task for subscription starts at T0. Say the tasks fails or rudder-sources restarts or some crash without fetching all records in first sync?
2. Next sync gets triggered - Would it recognize that essentially it has to get from subscriptions and not events because first sync did not complete successfully?

Lets discuss over a meeting if possible.